keps/sig-storage/0661-statefulset-volume-resize/README.md

```diff
@@ -0,0 +1,808 @@
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+This proposal does not try to address the following scenarios but relies on the
+underlying PVC resize logic to take the appropriate action or indicate error as needed.
+- Shrinking of volume is not allowed.
+- Few environments require Pod restarts in order
+to get access to the resized volume.
+- Modifying the storage class of the volumeClaimTemplate
+- Resizing a PVC associated with a statefulset directly by modifying the PVC spec will not affect the 
+  Statefulset `volumeClaimTemplates`. 
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Before we proceed to the proposal it is imperative to layout the various components involved
+in creating and managing the persistent volumes associated with `StatefulSet`. 
+
+- StatefulSet controller creates the PVCs from `volumeClaimTemplates` and associates them with the pods
+spawned from it. Reconciliation logic runs for every change made to the StatefulSet
+`Spec` and `Status` reflects the progress towards making the real world state equal to the
+`Spec`.
+- The admission controller in the API server is responsible for restricting the fields
+which can be edited in a StatefulSet specification.
+- The RBAC specification in the controller policy is responsible for adding the required 
+RBAC permission for StatefulSet.
+- The StatefulSet controller also supports revision control.
+
+The following changes are proposed to achieve the goal:
+1. Make the `volumeClaimTemplates` storage size editable by modification to the api server
+   validation. Some validation checks will depend on the design decisions made in [KEP 1790](../1790-recover-resize-failure/README.md) to prevent user from shrinking the volume.
+2. Add reconciliation of the associated PVC's size and the individual `volumeClaimTemplates` size into the 
+   StatefulSet controller reconciliation.
+3. Add PVC `update` privilege to the StatefulSet controller's clusterrole RBAC.
+
+Once the above changes are made, modification to any of the `volumeClaimTemplates` storage
+size by user will update the underlying PVC via the reconciliation performed by (2) above.
+The existing PVC resize workflow kicks in at this point and attempts to resize the volume.
+
+### Error indications to user
+
+Whether a resize works or not would depend on many factors such as is `allowExpansion` set in the
+storage class, does the underlying storage allow expansion etc. Such failures will be reported in
+the corresponding PVC events. StatefulSet would only report any failures in the update operation
+performed to the PVC object.
```

Comment:
If a resize error happens, does the control plane fix that? If not, what steps can someone with write access to a StatefulSet take in order to prompt a retry / ensure reconciliation of desired and actual state?

Reply:
Answers to that might belong in the Caveats section.

Reply:
Yes I think we need to clarify - what happens if user edits `volumeClaimTemplate` and increases PVC size but underlying volume plugin does not support expansion. Is the change blocked via admission or change is accepted but never applied?